### PR TITLE
Fix Java tracer library context propagation

### PR DESCRIPTION
**Pull request by PerfectSlayer**
### What does this PR do?

Update the Java tracer library context propagation documentation to use the latest settings available.

### Motivation

`DD_PROPAGATION_STYLE_INJECT` and `DD_PROPAGATION_STYLE_EXTRACT` are the old and now deprecated settings.
Moreover, they do not support all the propagation styles listed in the documentation page (`tracecontext` is missing) so this lead to confusion for customers.

### Additional Notes

---

### Reviewer checklist
- [x] Review the changed files.
- [x] Review the URLs listed in the [Preview](#preview) section.
- [x] Check images for PII
- [x] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by hestonhoffman — APPROVED** [review]
LGTM! Not sure if you need a review from @bantonsson before I merge, but ping me when you're ready for merge and I'll :shipit:

**Comment by PerfectSlayer** [comment]
He was the one in charge of the feature. I want to be sure the fix is right before landing into customers hands.

**Review by bantonsson — APPROVED** [review]

**Comment by PerfectSlayer** [comment]
@hestonhoffman Feel free to merge!

**Comment by bantonsson** [comment]
@kayayarai I think that the changes in #17518 might have reintroduced these deprecated settings into the docs for more languages than Java.

**Comment by PerfectSlayer** [comment]
I had the same feeling but I was not sure if the settings was unified for all languages or not.

**Comment by kayayarai** [comment]
Ok, I'll check. Thanks for raising this

**Pull request merged**
